cmd/sidecars/kafka_new/client/reader.go

```diff
@@ -0,0 +1,145 @@
+package kafkasc
+
+import (
+	"context"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	globalEnv "github.com/inspr/inspr/pkg/environment"
+	"github.com/inspr/inspr/pkg/ierrors"
+	"github.com/inspr/inspr/pkg/sidecars/models"
+	"go.uber.org/zap"
+)
+
+const pollTimeout = 100
+
+// Reader reads/commit messages from the channels defined in the env
+type Reader struct {
+	consumers map[string]models.Consumer
+}
+
+// NewReader return a new Reader
+func NewReader() (models.Reader, error) {
+	var reader Reader
+	channelsList := globalEnv.GetChannelBoundaryList(globalEnv.GetInputChannels())
+
+	resolvedChList := globalEnv.GetResolvedBoundaryChannelList(globalEnv.GetInputChannels())
+	if len(resolvedChList) == 0 {
+		return nil, ierrors.NewError().Message("INSPR_INPUT_CHANNELS not specified").InvalidChannel().Build()
+	}
+
+	reader.consumers = make(map[string]models.Consumer)
+
+	for idx, ch := range channelsList {
+		if err := reader.newSingleChannelConsumer(ch, resolvedChList[idx]); err != nil {
+			return nil, err
+		}
+	}
+	return &reader, nil
+}
+
+// Consumers returns a Reader's consumers
+func (reader *Reader) Consumers() map[string]models.Consumer {
+	return reader.consumers
+}
+
+/*
+ReadMessage reads message by message. Returns channel the message belongs to,
+the message and an error if any occurred.
+*/
+func (reader *Reader) ReadMessage(ctx context.Context, channel string) ([]byte, error) {
+	resolved, _ := globalEnv.GetResolvedChannel(channel, globalEnv.GetInputChannels(), "")
+
+	logger.Info("trying to read message from topic",
+		zap.String("channel", channel),
+		zap.String("resolved channel", resolved),
+	)
```

Comment:
i'm not sure why resolved channel is being created if he is only used for the logger